### PR TITLE
[Feature] Complete NCDA

### PR DESCRIPTION
**Pull request by anvmn**
#698

### CODE REVIEW

**Comment by anvmn** [comment]
Selection_1823

**Comment by anvmn** [comment]
@AronNovak please review

File: server/elm/src/Pages/Scoreboard/Utils.elm [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module Pages.Scoreboard.Utils exposing (..)
+
+
+tmp =
+    0
```
Comment (AronNovak) [inline server/elm/src/Pages/Scoreboard/Utils.elm]: What's the purpose of this? Do we use it somewhere?

File: server/hedley/modules/custom/hedley_aggregated_ncda/js/elm.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+   * Adds Elm application.
+   */
+  Drupal.behaviors.ElmApps = {
+    attach: function (context, settings) {
+      var elmApps = settings.elm_apps;
+      // Iterate over the apps.
+      Object.keys(elmApps).forEach(function (appName) {
+        // appName with unique css ID, e.g. `elm-app--2`.
+        var node = document.getElementById(appName);
+        if (!node) {
```
Comment (AronNovak) [inline server/hedley/modules/custom/hedley_aggregated_ncda/js/elm.js]: is it on purpose that we do not use jQuery for this? Also `context` and `once`?

File: server/elm/elm-watch.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  # --> all the errors you see are the current ones, not stale
+  clear;
+  tput reset;
+
+  echo -en "${DIM}";
+  date -R;
+  echo -en "${COLOR_OFF}";
+
+  elm make src/Main.elm --output ../hedley/modules/custom/hedley_aggregated_ncda/js/elm-main.js
+  # on Linux optionally prepend for better performance: sysconfcpus -n 1
```
Comment (AronNovak) [inline server/elm/elm-watch.sh:24]: We are able to conditionally invoke `sysconfcpus` if it's present.
Do you plan to use it inside DDEV or outside?

Comment (AronNovak) [reply server/elm/elm-watch.sh:24]:
```bash
if command -v sysconfcpus >/dev/null; then
    sysconfcpus -n 1 elm make src/Main.elm --output ../hedley/modules/custom/hedley_aggregated_ncda/js/elm-main.js
else
    elm make src/Main.elm --output ../hedley/modules/custom/hedley_aggregated_ncda/js/elm-main.js
fi

```

Comment (AronNovak) [reply server/elm/elm-watch.sh:24]: Perhaps we might want to install that tool in `.ddev/web-build/Dockerfile`

File: server/elm/src/Pages/Scoreboard/Utils.elm (hunk 1 above)
Comment (anvmn) [reply server/elm/src/Pages/Scoreboard/Utils.elm]: @AronNovak this PR lays infrastructure for the feature.
Right now, there're no utility funcitons, but very likely that we'll  need this file when developing the feature.
Elm does not allow empty file, so I put there something temporary for now.

File: server/hedley/modules/custom/hedley_aggregated_ncda/js/elm.js (hunk 2 above)
Comment (anvmn) [reply server/hedley/modules/custom/hedley_aggregated_ncda/js/elm.js]: @AronNovak 

Whole Elm app was ported from WFP-INCE.
Checked several other projects where we attach Elm apps, and we have this code at all.
Do you see a problem with current implementation?

Comment (AronNovak) [reply server/hedley/modules/custom/hedley_aggregated_ncda/js/elm.js]: the only thing is that `attached` have the potential to be called multiple times during the lifetime of a page. Could it be problematic? If not, I do not see any issues.

File: server/elm/elm-watch.sh (hunk 3 above)
Comment (anvmn) [reply server/elm/elm-watch.sh:24]: @AronNovak again, we got this from WFP-INCE.
On my local station current way of compiling works just fine.
It's very fast as well, so, I don't see a need for CPU optimizations right now.

File: server/hedley/modules/custom/hedley_aggregated_ncda/js/elm.js (hunk 2 above)
Comment (anvmn) [reply server/hedley/modules/custom/hedley_aggregated_ncda/js/elm.js]: Never seen functionality issue here, so, I don't think this method causes any problems.

**Review by AronNovak — APPROVED** [review]

File: server/hedley/modules/custom/hedley_ncda/scripts/generate-data-for-all.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
+<?php
+
+/**
+ * @file
+ * Generates new revisions for all entities of certain content type.
```
Comment (anvmn) [inline server/hedley/modules/custom/hedley_ncda/scripts/generate-data-for-all.php]:
```suggestion
 * Generates aggregated NCDA data for residents from all villages.
```

**Pull request merged**